File: sdk/typescript/src/resources/metrics.ts
import type { DogData } from '../client.js';
import type { MetricsUtxo, HolderConcentration, RealizedCap } from '../types.js';

export class MetricsResource {
  constructor(private client: DogData) {}

  async utxo(): Promise<MetricsUtxo> {
    return this.client.request<MetricsUtxo>('/api/metrics/utxo');
  }

  async holderConcentration(): Promise<HolderConcentration> {
    return this.client.request<HolderConcentration>('/api/metrics/holder-concentration');
  }

  async realizedCap(): Promise<RealizedCap> {
    return this.client.request<RealizedCap>('/api/metrics/realized-cap');
  }

  async all(): Promise<{
    utxo: MetricsUtxo;
    concentration: HolderConcentration;
    realizedCap: RealizedCap;
  }> {
    const [utxo, concentration, realizedCap] = await Promise.all([
      this.utxo(),
      this.holderConcentration(),
      this.realizedCap(),
    ]);
    return { utxo, concentration, realizedCap };
  }
}

File: sdk/python/dogdata/resources/metrics.py
from __future__ import annotations

from typing import Any, TYPE_CHECKING

if TYPE_CHECKING:
    from dogdata.client import DogData


class MetricsResource:
    def __init__(self, client: DogData):
        self._client = client

    def utxo(self) -> dict[str, Any]:
        return self._client.request("/api/metrics/utxo")

    def holder_concentration(self) -> dict[str, Any]:
        return self._client.request("/api/metrics/holder-concentration")

    def realized_cap(self) -> dict[str, Any]:
        return self._client.request("/api/metrics/realized-cap")

    def all(self) -> dict[str, Any]:
        return {
            "utxo": self.utxo(),
            "concentration": self.holder_concentration(),
            "realized_cap": self.realized_cap(),
        }
